concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransferScheduleTransaction.java

```diff
@@ -0,0 +1,113 @@
+package com.concordium.sdk.transactions;
+
+
+import com.concordium.sdk.exceptions.TransactionCreationException;
+import com.concordium.sdk.types.UInt64;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+import java.util.Objects;
+
+
+/**
+ * Construct a transaction to transfer an amount with schedule.
+ */
+@Getter
+public class TransferScheduleTransaction extends AbstractTransaction {
+    /**
+     * The account address of the recepient.
+     */
+    private final AccountAddress to;
+    /**
+     * The release schedule. This can be at most 255 elements.
+     */
+    private final Schedule[] schedule;
+
+    /**
+     * Account Address of the sender.
+     */
+    private final AccountAddress sender;
+    /**
+     * The senders account next available nonce.
+     */
+    private final AccountNonce nonce;
+    /**
+     * Indicates when the transaction should expire.
+     */
+    private final Expiry expiry;
+    /**
+     * A signer object that is used to sign the transaction.
+     */
+    private final TransactionSigner signer;
+
+    /**
+     * Maximum energy **allowed** for the transaction to use.
+     */
+    private final UInt64 maxEnergyCost;
```

Comment:
I don't think this should be here since the cost of the transaction can be calculated beforehand.
See https://github.com/Concordium/concordium-base/blob/75f015bb5f6483f5fdb5b49c789113a6f419038a/rust-src/concordium_base/src/transactions.rs#L2042

Reply:
Changed